### PR TITLE
Extract btsieve crate

### PR DESCRIPTION
**Pull request by thomaseizinger**
This time for real.

The overall motivation for this is to improve compile-times across our workspace, test things in isolation, allow for a narrower focus when debugging issues and potential reuse in other / future projects (we have re-written transaction monitoring a couple of times now).

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -12,6 +12,7 @@ members = [
   "xtra-bitmex-price-feed",
   "bdk-ext",
   "model",
+  "bitcoin-transaction-monitor",
```
Comment (da-kami) [inline Cargo.toml]:
```suggestion
  "btsieve",
```

🤪

**Review by da-kami — APPROVED** [review]
nice work!

File: Cargo.toml (hunk 1 above)
Comment (thomaseizinger) [reply Cargo.toml]: NOOOOOOOOOOOOOOO! :joy:

**Comment by thomaseizinger** [comment]
I don't understand why CI is failing, I tested opening a position locally and it worked like a charm.

File: Cargo.toml (hunk 1 above)
Comment (bonomat) [reply Cargo.toml]: 🤡

**Comment by klochowicz** [comment]
@thomaseizinger It seems to be failing in the smoke test, it might be just the case that it takes a bit of time to set up all the counters in the daemon. It should work if you increase the timeout on the smoke test!

**Comment by thomaseizinger** [comment]
I am doing a binary search of why CI is failing by force-pushing different sets of commits. Please ignore until I have found the problem!

**Comment by thomaseizinger** [comment]
> @thomaseizinger It seems to be failing in the smoke test, it might be just the case that it takes a bit of time to set up all the counters in the daemon. It should work if you increase the timeout on the smoke test!

But there are no counters in this PR :D

File: Cargo.toml (hunk 1 above)
Comment (thomaseizinger) [reply Cargo.toml]: Fine, long live btsieve! I'll change the name.

**Comment by klochowicz** [comment]
> > @thomaseizinger It seems to be failing in the smoke test, it might be just the case that it takes a bit of time to set up all the counters in the daemon. It should work if you increase the timeout on the smoke test!
> 
> But there are no counters in this PR :D

it still is just the smoke test, not the real tests ;D

**Comment by thomaseizinger** [comment]
> > > @thomaseizinger It seems to be failing in the smoke test, it might be just the case that it takes a bit of time to set up all the counters in the daemon. It should work if you increase the timeout on the smoke test!
> > 
> > 
> > But there are no counters in this PR :D
> 
> it still is just the smoke test, not the real tests ;D

And yet it is reproducible on CI! I don't understand :(

File: Cargo.toml (hunk 1 above)
Comment (klochowicz) [reply Cargo.toml]: `btc-sieve`?

**Comment by thomaseizinger** [comment]
> I am doing a binary search of why CI is failing by force-pushing different sets of commits. Please ignore until I have found the problem!

So I could reproduce that it is literally the last commit that introduces the problem ..

File: Cargo.toml (hunk 1 above)
Comment (thomaseizinger) [reply Cargo.toml]: Nice one @klochowicz :D

**Comment by thomaseizinger** [comment]
Issue seems to be resolved with the changes I am making in #1554. Will leave this as draft until the other on is fully-rebased and ready.

File: daemon/src/monitor.rs [hunk 2]
```diff
@@ -674,22 +675,18 @@ impl fmt::Display for ScriptStatus {
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Ord, PartialOrd)]
 struct BlockHeight(u32);
 
-impl fmt::Display for BlockHeight {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{}", self.0)
+impl From<usize> for BlockHeight {
+    fn from(height: usize) -> Self {
+        let height = u32::try_from(height)
+            .expect("bitcoin block count exceeds u32::MAX in > 80_000 years; qed");
```
Comment (luckysori) [inline daemon/src/monitor.rs]: 😆

File: btsieve/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+[package]
+name = "btsieve"
```
Comment (luckysori) [inline btsieve/Cargo.toml:2]: image

**Review by luckysori — APPROVED** [review]
Nice refactor!

**Pull request merged**